XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/HandJointController.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using XRTK.Definitions.Devices;
+using XRTK.Definitions.Utilities;
+using XRTK.Interfaces.InputSystem;
+using XRTK.Interfaces.Providers.Controllers;
+using XRTK.Services;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    public class HandJointController : BaseControllerDataProvider, IMixedRealityHandJointService
```

Comment:
Should a controller be inheriting from a data provider and a service?

Reply:
No it should not. Great catch. Maybe we want a `HandControllerDataProvider`?

Reply:
Agreed, But wouldn't we also have different providers for different platforms, e.g. HL2, Leap, ML ?

The critical thing is we have a generic way to expose those providers outputs in to a hand type controller?

Reply:
I think a BaseHandControllerDataProvider that extends BaseControllerDataProvider makes the most sense.

Reply:
> I think a BaseHandControllerDataProvider that extends BaseControllerDataProvider makes the most sense.

Especially if we need multiple hand data providers.